lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -2201,6 +2202,34 @@ bool QuakeBridgeVisitor::VisitInitListExpr(clang::InitListExpr *x) {
       return arrTy.getElementType();
     return initListTy;
   }();
+
+  if (initializerIsGlobal) {
+    static unsigned counter = 0;
+    auto *ctx = builder.getContext();
+    auto globalTy = cc::ArrayType::get(ctx, eleTy, size);
+    SmallVector<Attribute> values;
+    auto f64Ty = builder.getF64Type();
+    for (Value v : last) {
+      auto fp = opt::factory::maybeValueOfFloatConstant(v);
+      assert(fp);
+      values.push_back(FloatAttr::get(f64Ty, *fp));
```

Comment:
We do want these values to be of ComplexType. You could see a simple float, but it should represent `FloatVal + 0.0 * i`.

Reply:
For this PR, I'm made the assumption (probably bogus) that the initializer list is a $2^n$ list of floats and the pairwise complex values and just even and odd offsets in that list.

If we want to allow fancier complex literals, we'll have to extend this.

Reply:
The state-vector is always a complex-valued entity. But there are likely use cases where the user would like to skirt around having to specify the full complex representation for the elements because the state is real-valued. Hence the `qvector` constructors on both `std::vector<complex<double>>` and `std::vector<double>` (we also will need `std::vector<float>` and `std::vector<complex<float>>`). I imagine bridge handling for complex elements is harder than what we have here, but we will need it. Do you want to do that in a follow up PR or here?

Reply:
Let's make it a follow-up. There is a complex builtin type, but the calling conventions for complex values are mysterious. That'll take some extra work.